test/examples/test_delivery_mechanisms.py

```diff
@@ -0,0 +1,159 @@
+# Copyright 2024 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import subprocess
+import pytest
+
+delivery_test_cases = [
+    # Default builtin transports (remove isolated subscriber to avoid flakiness)
+    ('', '-s 20', '--unknown-argument', '', 30),
+    ('', '-s 20', '--unknown-argument', '--ignore-local-endpoints', 30),
+    # Data-sharing isolated subscriber tested in timeout test cases
+    ('--mechanism data-sharing', '-s 20 --mechanism data-sharing', '--unknown-argument', '--mechanism data-sharing', 30),
+    ('--mechanism data-sharing', '-s 20 --mechanism data-sharing', '--unknown-argument', '--mechanism data-sharing --ignore-local-endpoints', 30),
+    # Intra-process only makes sense for pubsub entities. This test forces entities != pubsub  to fail, so
+    #  only 1 message is expected per sample (the local one)
+    ('--unknown-argument', '--unknown-argument', '--unknown-argument', '--mechanism intra-process', 10),
+    # Large-data (using shared memory, so isolated subscriber is removed to avoid flakiness)
```

Comment:
Why do we only test using shared memory and not UDP+TCP?

Reply:
Because by default it uses SHM (net host, icp host, same container...)
Only when explicitly isolating each endpoint, they will communicate over TCP (check isolated tests and isolated compose)